index.ts

```diff
@@ -1,9 +1,43 @@
 import { Template } from './workspace/template'
+import * as express from 'express'
+import { Result } from './workspace/result'
+import { Bus } from './workspace/bus'
+import * as cors from 'cors'
 
 export class Index {
     public static main(): number {
-        const template = new Template();
-        template.run();
+        var app = express();
+        app.use(cors());
+        app.get("/", function (req, res){
+            res.send('Hello World!')
+        });
+        var urlString: string;
+        app.get('/closestStops', function (req, res){
+            const template = new Template();
+            var postcodeRegex = new RegExp("[a-zA-Z]?[a-zA-Z][0-9]?[0-9][0-9][a-zA-Z][a-zA-Z]$");
+            var booleanPostcode = postcodeRegex.test(req.query.postcode)
+            if (booleanPostcode==true){
+                var resultsArrayPromise : Promise <Result[]> = template.run(req.query.postcode);
+                var resultString: string = ""; 
+                resultsArrayPromise.then((resultsArray) => {
+                    if (resultsArray.length != 0){
+                        res.status('200')
+                        res.send(resultsArray);
+                    }
+                    else {
+                        res.send('No buses found')
+                        res.status('400')
+                    }
+                })
+            }
+            else {
+                res.status('406')
```

Comment:
I'm really happy that you've thought to but in error codes! That makes this handler a lot nicer.

I think 406 is the wrong code, though - [looks like it's to do with accept errors](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#4xx_Client_errors). Sadly, most of the time, the correct code is boring old 400, if the input is bad.

(If you ever find a reason to use code 418, please let me know)